### PR TITLE
docs: openai monitoring readme

### PR DESCRIPTION
**Pull request by staceysv**
* add a super minimal REAMDE.md in examples/monitoring
* summarize OpenAI authentication via API/SDK vs proxy
* describe and link to list of demo notebooks—including proxy quickstart which is not yet merged
* move dev notebooks into dev :)

TODO:
- [x] do we want to link to monitor_api notebook? if yes should we clean it up to be more usable? or just put in dev?
- [x] make sure proxy quickstart works
- [x] add pretty overview gif to readme?

### CODE REVIEW

File: examples/monitoring/README.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+See details and create an interactive board in the [OpenAI monitoring notebook](../monitoring/openai_ux_demo.ipynb).
+
+### OpenAI Proxy
+
+To log all calls made via an OpenAI proxy:
+
+1. Find [your wandb API key](https://wandb.ai/authorize) and [your OpenAI API key](https://platform.openai.com/account/api-keys).
+2. Set your OPENAI_API_KEY to these two keys joined by ":", i.e. wandb_api_key:openai_api_key.
+3. Change the OpenAI base url to `https://wandb.ai/proxy/openai/v1`.
+4. Optionally set headers on each call to configure the wandb entity, project, and data stream name.
```
Comment (jlzhao27) [inline examples/monitoring/README.md:30]: can we add a markdown table with all the supported options and headers (ignore my syntax):

```
header name | description | example
X-Wandb-Project | sets the wandb project the results will be logged to |  -H "X-Wandb-Project: chat-completion"
```

**Comment by staceysv** [comment]
- [ ] test proxy on local
- [ ] confirm netrc compatibility
- [ ] merge https://github.com/wandb/weave/pull/391
- [ ] alignment on main demo examples vs dev notebooks
- [ ] entity and project setting in headers
- [ ] clean urls to data streams/correct config for this
- [ ] proxy set agent

File: examples/monitoring/README.md [hunk 2]
```diff
@@ -30,7 +33,7 @@ To log all calls made via an OpenAI proxy:
 2. Find [your OpenAI API key](https://platform.openai.com/account/api-keys).
 3. Set your OPENAI_API_KEY to these two keys joined by ":", i.e. wandb_api_key:openai_api_key.
 4. Change the OpenAI base url to `https://wandb.ai/proxy/openai/v1` for the public cloud instance (or replace `wandb.ai` with the base url for your instance).
-5. Optionally set headers on each call to customize the wandb entity, project, data stream name, and other aspects of logging (see the table below for details).
+5. Optionally set headers on each call to customize the wandb entity, project, and other aspects of logging (see the table below for details).
```
Comment (staceysv) [inline examples/monitoring/README.md:36]: I think it's important to call out that we can name/individually manage data streams here.

File: examples/monitoring/openai_ux_demo.ipynb [hunk 3]
```diff
@@ -8,11 +8,12 @@
     "# OpenAI LLM Monitoring with W&B Weave\n",
     "\n",
     "Use the W&B OpenAI integration to monitor OpenAI API calls and understand how your projects and teams are leveraging LLMs.\n",
-    "This example notebook will generate links to Weave Boards: LLM usage monitoring dashboards which you can explore and customize from the UI.\n",
+    "In this example, we'll setup a LLM usage monitoring dashboards which you can explore and customize from the UI.\n",
```
Comment (staceysv) [inline examples/monitoring/openai_ux_demo.ipynb:11]: compromise: In this example, we'll generate templated Weave Boards: ?

File: examples/monitoring/openai_ux_demo.ipynb [hunk 4]
```diff
@@ -8,11 +8,12 @@
...
     " \n",
+    "* track cost, latency and token counts to see how your organization are using LLMs\n",
+    "* manage permissions and hosting to ensure data privacy\n",
```
Comment (staceysv) [inline examples/monitoring/openai_ux_demo.ipynb:14]: took a pass at a compromise, I really don't think this "manage permissions and hosting to ensure data privacy" is true/possible right now.

**Review by scottire — APPROVED** [review]

**Pull request merged**